Python is a powerful tool developers use with one of its most desireble capabilities being data processing. In this Colab Notebook you will run through an example of how Python can be used to perform simple data analysis on an airline's customer data. 

1. First we will need to download loacally and then upload the [AirlinesCluster.csv](https://drive.google.com/file/d/1zQeqXkJ5Eb30MBqAe0VVjRaEmSUrYF1j/view?usp=sharing) file to Google Colab. We can then read the AirlinesCluster.csv file and create a Pandas dataframe object. This object will have 6 columns being Balance, BonusMiles, BonusTrans, FlightMiles, and DaysSinceEnroll. 

In [ ]:
import pandas as pd
from google.colab import files
import io
 
uploaded = files.upload()

df = pd.read_csv(io.BytesIO(uploaded['AirlinesCluster.csv']))


Saving AirlinesCluster.csv to AirlinesCluster (1).csv


2. Let's display the data to the console by calling the dataframe we just created. 


In [ ]:
df

Balance  BonusMiles  BonusTrans  FlightMiles  FlightTrans  \
0       48296       31329           9          500            1   
1       10021           0           0            0            0   
2       49280       22370          16            0            0   
3      213539        2750          15            0            0   
4      125465       14750           9            0            0   
...       ...         ...         ...          ...          ...   
3994     3016           0           0            0            0   
3995    28577       48564          14            0            0   
3996   276571       42044          23            0            0   
3997    28848           0           0            0            0   
3998    96522       61105          19            0            0   

      DaysSinceEnroll  TotalMiles  
0                3061       31829  
1                7879           0  
2                3312       22370  
3                4751        2750  
4                7206       14750  
...               ...         ...  
3994             1398           0  
3995             3586       48564  
3996             7872       42044  
3997             3069           0  
3998             6924       61105  

[3999 rows x 7 columns]

3. We can see that there are 3999 datapoints. Let's next find the mean, or average, on all of the customers' accounts.

In [ ]:
df.mean()

Balance            73601.327582
BonusMiles         17144.846212
BonusTrans            11.601900
FlightMiles          460.055764
FlightTrans            1.373593
DaysSinceEnroll     4118.559390
TotalMiles         17604.901975
dtype: float64

4. We can also find the sum of each customer's bonus miles and flight miles

In [ ]:
df['TotalMiles'] = df["BonusMiles"] + df['FlightMiles']
df

Balance  BonusMiles  BonusTrans  FlightMiles  FlightTrans  \
0       48296       31329           9          500            1   
1       10021           0           0            0            0   
2       49280       22370          16            0            0   
3      213539        2750          15            0            0   
4      125465       14750           9            0            0   
...       ...         ...         ...          ...          ...   
3994     3016           0           0            0            0   
3995    28577       48564          14            0            0   
3996   276571       42044          23            0            0   
3997    28848           0           0            0            0   
3998    96522       61105          19            0            0   

      DaysSinceEnroll  TotalMiles  
0                3061       31829  
1                7879           0  
2                3312       22370  
3                4751        2750  
4                7206       14750  
...               ...         ...  
3994             1398           0  
3995             3586       48564  
3996             7872       42044  
3997             3069           0  
3998             6924       61105  

[3999 rows x 7 columns]

5. Another example of Python's capabilities is being able to find the mode of an entire dataset. As a reminder, mode is a mathemetical term meaning the most repeated value. Below we will take the mode of each column in the dataset. 

In [ ]:
df.mode()

Balance  BonusMiles  BonusTrans  FlightMiles  FlightTrans  DaysSinceEnroll  \
0     1000           0           0            0            0             8296   

   TotalMiles  
0           0

6. Likewise, we can easily find he median of all the columns 

In [ ]:
df.median()

Balance            43097.0
BonusMiles          7171.0
BonusTrans            12.0
FlightMiles            0.0
FlightTrans            0.0
DaysSinceEnroll     4096.0
TotalMiles          7747.0
dtype: float64